src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -503,11 +507,17 @@ private void insert(final SinkRecord record)
         }
         else
         {
+          // lag telemetry, note that sink record timestamp might be null
+          if (snowflakeRecord.timestamp() != null &&
```

Comment:
Can snowflake record be null ?

Reply:
No. snowflakeRecord is guarantied to be not null.